### PR TITLE
facets: integrate subjects faceting

### PR DESCRIPTION
**Pull request by fenekku**
- pre-requisite: https://github.com/inveniosoftware/invenio-rdm-records/pull/790
- closes https://github.com/inveniosoftware/invenio-vocabularies/issues/68﻿

facets_subjects_2

### CODE REVIEW

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search.html [hunk 1]
```diff
@@ -40,8 +40,13 @@
...
     {
       "aggName": "subjects",
-      "field": "subjects",
-      "title": "{{_("Subjects")}}"
+      "field": "subjects.scheme",
+      "title": "{{_("Subjects")}}",
+      "childAgg": {
+        "aggName": "inner",
+        "field": "subjects.subject",
+        "title": "{{_("Resource types")}}"
```
Comment (ppanero) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search.html:48]: This title is a typo right?
```suggestion
        "title": "{{_("Subjects")}}"
```

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search_deposit.html [hunk 2]
```diff
@@ -40,8 +40,13 @@
...
     {
       "aggName": "subjects",
-      "field": "subjects",
-      "title": "{{_("Subjects")}}"
+      "field": "subjects.scheme",
+      "title": "{{_("Subjects")}}",
+      "childAgg": {
+        "aggName": "inner",
+        "field": "subjects.subject",
+        "title": "{{_("Resource types")}}"
```
Comment (ppanero) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search_deposit.html:48]: ditto

**Comment by lnielsen** [comment]
I'm thinking that given that NYU reported people having difficulties with the nested facets, it's better not to nest in the case of subjects, but instead a simple plain list like:

- Coronavirus, Feline (MeSH)
- Taxis Response (MeSH)
- ...

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search.html (hunk 1 above)
Comment (fenekku) [reply invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/search.html:48]: Yep thanks! (I don't know where it shows up on the UI :shrug: )

**Comment by fenekku** [comment]
See reply here: https://github.com/inveniosoftware/invenio-rdm-records/pull/790#discussion_r681738782 and ticket here: https://github.com/inveniosoftware/invenio-app-rdm/issues/1043

**Comment by lnielsen** [comment]
Not needed anymore as the config is now generated.

**Pull request closed without merging**